app/Models/Album.php

```diff
@@ -183,6 +185,9 @@ public function delete(): bool
 		try {
 			$this->refreshNode();
 
+			// delete track
+			(new Tracks())->delete($this->id);
```

Comment:
This does only delete the tracks of the current album, but does not delete the tracks of sub-albums.

Reply:
How are they deleted? It seems that this `delete` method is never called for subalbums, do they use another method or do I have to fetch all the subalbums and run `deleteTrack` separately on them?

Reply:
> or do I have to fetch all the subalbums and run deleteTrack separately on them?

Basically this it is. Unfortunately, it is completely inefficient due to Laravels insance architecture. We have the same problem when we need to delete the photos and symbolic links, when an album is deleted. That is why we fetch all photos  via `->all_photos` some lines above.

However, I have another suggestion. The existing code (not your,s our code some lines below) is architecturally correct, but causes timeouts when you delete large sub-trees with many photos in it. (Because it hydrates all the photos within.) I already planned to fix that. I will fix that part and then you put your code to delete tracks on top of it.

Just give me some days to do it.

Reply:
@qwerty287 This is just a heads up. The PR https://github.com/LycheeOrg/Lychee/pull/1277 cleans up the deletion of media files and symlinks for current master. After this has been reviewed, approved and merged into the master branch, I will port it to `consistent_json_api`. Then you can add the code for deleting tracks.

Just in case you already would like to have a look.

Reply:
Thanks! I'll look into it once it is ported.